src/main/java/ui/MenuControl.java

```diff
@@ -89,47 +87,71 @@ private Menu createFileMenu() {
     }
 
     /**
-     * Logs the user action and creates a new board
+     * Checks whether the current open board is dirty, i.e. has unsaved changes.
+     *
+     * @return true if current open board is dirty, false otherwise
      */
-    private void onBoardNew() {
-        logger.info("Menu: Boards > New");
-        newBoard();
+    public final boolean isCurrentOpenBoardDirty() {
+        if (!prefs.getLastOpenBoardPanelInfos().isPresent()) {
+            return true;
+        }
+        return !panels.getCurrentPanelInfos().equals(prefs.getLastOpenBoardPanelInfos().get());
     }
 
     /**
-     * Creates an empty board. User will be prompted to
-     * confirm the action if there are unclosed panels
+     * Prompts a dialog to ask the user whether or not to save the current board.
      */
-    public final void newBoard() {
-        if (!isNewBoardCreationDialogConfirmed()) {
-            logger.info("New board creation cancelled");
+    public final void promptToSaveCurrentBoard() {
+        boolean shouldSave = DialogMessage.showYesNoConfirmationDialog("Save Changes?",
+                "There are unsaved changes to your current board.",
```

Comment:
I don't think the message should be this.

While it might be true, this will need us to check if board is dirty before calling this method. The line below this (All unsaved changes will be discarded) should be enough.